sds/src/scanner/regex_rule/regex_cache_store.rs

```diff
@@ -1,24 +1,27 @@
 use crate::scanner::regex_rule::regex_store::{RegexCacheKey, SharedRegex};
 use crate::SharedPool;
 use lazy_static::lazy_static;
-use regex_automata::meta::Regex as MetaRegex;
+use regex_automata_fork::meta::Regex as MetaRegex;
 use slotmap::SecondaryMap;
 use std::sync::Arc;
+extern crate num_cpus;
 
 lazy_static! {
-    static ref REGEX_CACHE_STORE: Arc<SharedPool<Box<RegexCaches>>> =
-        Arc::new(SharedPool::new(|| Box::new(RegexCaches::new())));
+    static ref REGEX_CACHE_STORE: Arc<SharedPool<Box<RegexCaches>>> = Arc::new(SharedPool::new(
+        Box::new(|| Box::new(RegexCaches::new())),
+        num_cpus::get()
+    ));
 }
 
 pub fn access_regex_caches<T>(func: impl FnOnce(&mut RegexCaches) -> T) -> T {
     // This function isn't strictly necessary, but it makes it easier to change the implementation
     // later
     let mut caches = REGEX_CACHE_STORE.get();
-    func(&mut caches)
+    func(caches.get_ref())
 }
 
 pub struct RegexCaches {
-    map: SecondaryMap<RegexCacheKey, regex_automata::meta::Cache>,
+    map: SecondaryMap<RegexCacheKey, regex_automata_fork::meta::Cache>,
```

Comment:
Unfortunately rust complains in all directions with something like - the names match but the types are distinct - are these different versions of the same crate?

Reply:
This is because you didn't remove it in enough places. I pushed a [commit](https://github.com/DataDog/dd-sensitive-data-scanner/pull/140/commits/7e74d965074383785e289372e52b8e127d02f8c9) with this change